File: Day 2/Part 1.php
<?php
// Luke Sheppard
// Advent of Code 2019 - Day 2 - Part 1 - https://adventofcode.com/2019/day/2

$input = file_get_contents(__DIR__ . '/input.csv');

$opcodes = explode(',', $input);

for ($i = 0; $i < count($opcodes); $i += 4) {
    $opcode = $opcodes[$i];
    $params = array_slice($opcodes, $i + 1, 3);

    switch($opcode) {
        case 99: // Halt
            break 2;

        case 1: // Add
            $opcodes[$params[2]] = $opcodes[$params[0]] + $opcodes[$params[1]];
            break;

        case 2: // Multiply
            $opcodes[$params[2]] = $opcodes[$params[0]] * $opcodes[$params[1]];
            break;
    }
}

echo "Answer: " . $opcodes[0] . "\n";

File: Day 2/Part 1.rb
opcodes = File.open(__dir__ + '/input.csv').read.split(",").map(&:to_i)

opcodes.each_slice(4).with_index { |slice, index|
    opcode = slice[0]

    break if opcode == 99

    case opcode
    when 1
        opcodes[slice[3]] = opcodes[slice[1]] + opcodes[slice[2]]
    when 2
        opcodes[slice[3]] = opcodes[slice[1]] * opcodes[slice[2]]
    end
}

puts "Answer: #{opcodes[0].to_s}"
